src/continuous-validation/solid-solid-collision.cc

```diff
@@ -206,6 +222,38 @@ namespace hpp {
 
           ++i;
         }
+
+        // Compute coefficients going from joint b to joint a
+        if(joint_b)
+        {
+          JointIndices_t joints_reverse(joints);
+          std::reverse(joints_reverse.begin(),joints_reverse.end());
+          assert (joints_reverse.size () > 1);
+          coefficients_reverse.resize (joints_reverse.size () - 1);
+          robot =joint_b->robot ();
+          // Store r0 + sum of T_{i/i+1} in a variable
+          cumulativeLength = joint_b->linkedBody ()->radius ();
+          i = 0;
+          while (i + 1 < joints_reverse.size()) {
+            if (model.parents[joints_reverse[i]] == joints_reverse[i+1])
+              child = Joint::create (robot, joints_reverse[i]);
+            else if (model.parents[joints_reverse[i+1]] == joints_reverse[i])
+              child = Joint::create (robot, joints_reverse[i+1]);
+            else
+              abort ();
+            assert(child);
+            coefficients_reverse [i].joint_ = child;
+            // Go through all known types of joints
+            //  TODO: REPLACE THESE FUNCTIONS WITH NEW API
+            distance = child->maximalDistanceToParent ();
+            coefficients_reverse [i].value_ =
+              child->upperBoundLinearVelocity () +
+              cumulativeLength * child->upperBoundAngularVelocity ();
+            cumulativeLength += distance;
+
+            ++i;
+          }
+        }
```

Comment:
I would find better to have only one algo that computes weights for a sequence of joints. Is this easy to achieve ?

Reply:
Yes, I made the change.

Reply:
I fixed a bug in the computations of the coefficients when joint_b is the universe. It seems that I pushed to your branch.
I did not succeed in making a "suggested change".